### DISCUSSION TITLE
Force static HTML export

### DISCUSSION THREAD

**Discussion by kr8n3r** [vercel/next.js#13909] [Help] [answered]
Hi 👋 
we're looking to build performant product pages using mdx, so looking at static HTMl export.
However, running `next build && next export`, outputs as SSG
<img width="658" alt="Screenshot 2020-06-08 at 19 22 53" src="https://user-images.githubusercontent.com/3758555/84066233-82682d80-a9bd-11ea-8aba-98c9df21c4d9.png">

which is less than ideal, as we'd need to include legacy IE polyfills.

Any ideas what config we'd be missing?

Link to repo: https://github.com/alphagov/paas-product-pages

**Comment by Timer** [comment, marked as answer]
Not sure I understand your question. SSG and `next export` are nearly analogous!

`next export` loses some benefits though, i.e. Incremental Static (Re)Generation.

**Reply by kr8n3r** [reply]
Hi, if I rephrase: Plain HTML output that doesn't rely on HTML5 history-api (that SSG needs) for legacy IE

**Reply by Timer** [reply]
Aah, you're requesting an application that does not become interactive or execute any JavaScript?

**Reply by kr8n3r** [reply]
yep. output of plain old html pages that link together, static

**Reply by Timer** [reply]
You can disable JS generation on a per-page basis by exporting this from your page(s):
```js
export const config = {
  unstable_runtimeJS: false
}
```

How old of IE do you need to support? Next.js supports IE11 out of the box.

**Reply by kr8n3r** [reply]
cool, will try that. as pages are markdown,  I can export this in the layout?
support down to IE8

**Reply by Timer** [reply]
No, it needs to be exported in each mdx file.

**Reply by kr8n3r** [reply]
ok thanks, will experiment with that. generally, what would be condition under which export would export static HTML (as per CLI screenshot above) instead of HTML+JSON?

**Reply by Timer** [reply]
The JSON will not be used without client-side JS, making SSG as "dumb" as Static.

**Reply by kr8n3r** [reply]
so nextJS framework requires history api to work properly?
trying to understand if we're not using intentionally `initial props`, can we not just export plain html

**Reply by Timer** [reply]
Next.js requires the history API for **client-side transitions** to work: the same as any JS framework. This is what the JSON file is used for.

Since you want to disable JavaScript for IE8 support, every transition will be server-side, meaning your visitors will only be getting HTML.

I'd suggest going through the [Next.js tutorial](https://nextjs.org/learn) to learn how Next.js works, as it answers a lot of your questions.

**Reply by kr8n3r** [reply]
appreciate your help and as I understand, to trully have a no-js running site, the config you provided is the only way to achieve that?

**Reply by Timer** [reply]
The config I provided is the only way to run a no-JS site, correct.

You could always manually post-process the `out/` directory to remove all `<script>` tags, too.

**Reply by kr8n3r** [reply]
thanks @Timer
